### ISSUE TITLE
use DOT as native parachain currency and allow downward transfers

### ISSUE THREAD

**Issue by brenzi** [encointer/encointer-parachain#4]
scenario:
1. Start with zero DOTs on the parachain
2. send downward transfer X relay chain DOT to parachain
3. send tx on parachain, paying fees in DOT
4. send DOT back to relay chain

resources
* https://wiki.polkadot.network/docs/en/build-parachains-rococo#how-to-make-cross-chain-transfers

**Comment by brenzi** [comment]
This functionality seems not to be implemented for rococo-v1. We should wait until we have docs and examples from parity

**Comment by clangenb** [comment]
The current status for rococo-v1 is:

1. This is not wrapped in the polkadot-ui as it was for rococo-v0
2. Only available via sudo calls
3. A generic xcm message needs to be entered in a tedious manner.

**Comment by clangenb** [comment]
We decided to postpone this. There is a wip upward transfer pallet that wraps the generic xcm message in:

https://github.com/encointer/pallets/tree/pallet-xc-transfer-rococo

which has been integrated in: https://github.com/encointer/encointer-parachain/tree/native-dot-upward-and-downward-transfer

Current status: The xcm message is executed successfully. The balance is deducted from the sender, but it does not arrive at the relay chain.

**Comment by brenzi** [comment]
blocking upstream dependency:
https://github.com/paritytech/polkadot/issues/2564

**Comment by clangenb** [comment]
It does somewhat help us resolve this. Teleporting assets is currently limited to trusted teleporters, i.e. origins that are trusted to have irrevocably burned the assets in the origin system. These trusted teleporters are currently hard-coded in the runtime:

https://github.com/paritytech/polkadot/blob/master/runtime/rococo/src/lib.rs#L608-L613

**Comment by brenzi** [comment]
yes, do that plz! I suggest we don't wait as this issue is a strict necessity to be accepted as a common good parachain

**Comment by clangenb** [comment]
I believe we should wait after all, as the feature seems to be merged very soon, and it seems that the procedure for native up/-downward transfer is different for untrusted chains.

See: https://github.com/paritytech/polkadot/pull/3010.

**Comment by brenzi** [comment]
that PR is included in 0.9.2
https://github.com/paritytech/polkadot/commits/v0.9.2

**Comment by brenzi** [comment]
@clangenb could we do this now based on polkadot 0.9.8?

**Comment by clangenb** [comment]
We should be able to do this. Note: polkadot-v0.9.9 is released already (at least in the Polkadot repository).

Because we are (most-likely) going to become a trusted common good parachain, the approach will be simplified compared to above. We can directly transfer funds with `Teleportation` instead of a `ReserveAssetDeposit`-XCM xt.

**Comment by brenzi** [comment]
No. Unlike Statemine, our code is not trusted, as Encointer has its own governance, partially. So we can't use teleport

**Comment by clangenb** [comment]
Ok, you know that probably better than I do, but I thought it is a requirement to be trusted to become a common good parachain.

**Comment by brenzi** [comment]
Not strictly. Encointer will be the first exception to that rule I hope.

**Comment by brenzi** [comment]
polkadot and cumulus 0.9.9 are released. we could move forward with this

**Comment by clangenb** [comment]
We need 0.9.10

**Comment by brenzi** [comment]
there we go
https://github.com/paritytech/cumulus/releases/tag/polkadot-v0.9.10

**Comment by brenzi** [comment]
check out Shawn's XCM workshop on reserve transfer: https://www.shawntabrizi.com/xcm-workshop/#/reserve-transfer
